dockertest/dockercmd.py

```diff
@@ -334,6 +334,49 @@ def execute(self, stdin=None):
                                          stdin=stdin, close_fds=True)
         return self.cmdresult
 
+    def wait_for_ready(self, timeout=None, timestep=0.2):
+        """
+        Monitor the output of a container (including docker logs, in
+        case stdout is detached), waiting for the string 'READY'. Return
+        as soon as we see it. If we don't, throw a meaningful exception.
+
+        :raises DockerExecError: on timeout.
+        """
+        end_time = time.time() + timeout
+        while time.time() <= end_time:
+            time.sleep(timestep)
+            if 'READY' in self.stdout:
+                return
+            # Also check docker logs
+            cid = self.container_id
+            if cid is not None:
+                logs = DockerCmd(self.subtest, 'logs', [self.container_id])
+                logs.execute()
+                if 'READY' in logs.stdout:
+                    return
+
+        # Never saw READY. Did container exit? If so, help user understand why
+        if self.done:
```

Comment:
I need to think about this. The function is `wait_for_ready()`. It isn't `wait_for_ready_and_make_sure_the_container_is_still_running()`. All we do is check for the string READY. At present the only invocations also expect the container to continue running, but that may not always be the case. I will add `and not self.done` to the while loop, though, to avoid wasting time.